### PR TITLE
Add support of topic ids for the OffsetFetch API from version 9.

### PR DESCRIPTION
**Pull request by Hangleton**
WIP.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by jolshan** [comment]
Let me know if you have any topic ID questions @Hangleton

**Comment by Hangleton** [comment]
Hi, Justine (@jolshan), thanks for your help. There is another PR with the same purpose as this one for the `OffsetCommit` API: [PR-13240](https://github.com/apache/kafka/pull/13240). Please feel free to review. Happy to discuss further. Thanks!

File: clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchRequest.java [hunk 1]
```diff
@@ -253,38 +284,15 @@ public List<OffsetFetchRequestData.OffsetFetchRequestGroup> groups() {
         }
     }
 
-    public Map<String, List<TopicPartition>> groupIdsToPartitions() {
```
Comment (Hangleton) [inline clients/src/main/java/org/apache/kafka/common/requests/OffsetFetchRequest.java:256]: Moved to tests.

**Comment by dajac** [comment]
Closing this PR for now as the topic id work will be done later. We can re-open it when we resume the work.

**Pull request closed without merging**